### ISSUE TITLE
Issue when using custom component as ref

### ISSUE THREAD

**Issue by DennisOng** [ycs77/headlessui-float#23]
**Use Version**
Use version when bugs appear:
 - Headless UI: v1.7.0
 - Headless UI Float: v0.9.1
 - Framework: vue v3.2.31
 - Browser: chrome

**Describe the bug**
FloatingUI is not working when a custom component is used as ref, eg:
```
// Not working
        <PopoverButton as="template">
            <BaseButton>Button </BaseButton>
        </PopoverButton>

// Works
        <PopoverButton as="template">
             <button>Button</button>
        </PopoverButton>
```

**To Reproduce**
Please provide a minimal working example (like github repo, codesandbox, stackblitz...), and steps to reproduce the behavior:
1. Go to '[this StackBlitz](https://stackblitz.com/edit/github-9qim3c?file=src/components/ExamplePopover.vue)'
2. Click on the first button. It works well
3. Click on the second button. 
5. See console error. Also floating UI is not positioning correctly
```
TypeError: element.getBoundingClientRect is not a function
    at getBoundingClientRect (floating-ui.dom.esm.js:102:30)
```

**Screenshots**
<img width="628" alt="Capture" src="https://user-images.githubusercontent.com/2383712/191971510-5d3ee92c-0f2b-48c4-a4ea-2e7cad4b0e62.PNG">

**Expected behavior**
Button 2 should work exactly as button 1

PS: Nice work on this lib! Have spent way too much time trying to roll my own, switching over currently. Thanks for the work.

**Comment by ycs77** [comment]
Fixed in v0.10.0

**Comment by DennisOng** [comment]
Thanks for your work! Just tested with the latest release - issue has been resolved
